libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -531,24 +541,42 @@ internal class MapRouteLine(
      * @param directionsRoutes the routes to be represented on the map.
      */
     fun draw(directionsRoutes: List<DirectionsRoute>) {
-        val featureDataProvider: () -> List<RouteFeatureData> =
-            getRouteFeatureDataProvider(directionsRoutes)
-        reinitializeWithRoutes(directionsRoutes, featureDataProvider)
-        drawRoutes(routeFeatureData)
+        ThreadController.getMainScopeAndRootJob().scope.launch {
```

Comment:
Let's avoid adding asynchronous code to the `MapRouteLine` at this point. While I think it's a good idea and we should definitely explore it, I'd rather see it done in a separate PR since there are many more considerations than just offloading the work. As an example, the current code might easily run into a race condition when `draw` is called in quick succession with 2 different parameters which will result in displaying an outdated route.